src/Elmah.Io.AspNetCore/ElmahIoMiddleware.cs

```diff
@@ -8,44 +8,54 @@ namespace Elmah.Io.AspNetCore
     public class ElmahIoMiddleware
     {
         private readonly RequestDelegate _next;
+        private readonly string _apiKey;
+        private readonly Guid _logId;
         private readonly ElmahIoSettings _settings;
 
-        public static string ApiKey { get; private set; }
-        public static Guid LogId { get; private set; }
-
+        [Obsolete]
         public ElmahIoMiddleware(RequestDelegate next, string apiKey, Guid logId) : this(next, apiKey, logId, new ElmahIoSettings())
         {
         }
 
+        [Obsolete]
         public ElmahIoMiddleware(RequestDelegate next, string apiKey, Guid logId, ElmahIoSettings settings)
         {
             _next = next;
-            ApiKey = apiKey.AssertApiKey();
-            LogId = logId.AssertLogId();
+            _apiKey = apiKey.AssertApiKey();
+            _logId = logId.AssertLogId();
             _settings = settings.AssertSettings();
         }
 
+        public ElmahIoMiddleware(RequestDelegate next)
```

Comment:
Inject your configuration here instead of resolving it explicitly on every request.

In the other constructors, create the configuration object and store it locally, so you don’t need to check multiple sources below.